src/Elastic.Apm/Config/AbstractConfigurationReader.cs

```diff
@@ -430,6 +430,32 @@ protected double ParseTransactionSampleRate(ConfigurationKeyValue kv)
 			return parsedValue;
 		}
 
+		protected int ParseTransactionMaxSpans(ConfigurationKeyValue kv)
+		{
+			if (kv?.Value == null)
+			{
+				Logger?.Debug()
+					?.Log("No transaction max spans provided. Defaulting to '{DefaultTransactionMaxSpans}'",
+						ConfigConsts.DefaultValues.TransactionMaxSpans);
+				return ConfigConsts.DefaultValues.TransactionMaxSpans;
+			}
+
+			if (int.TryParse(kv.Value, NumberStyles.Integer, CultureInfo.InvariantCulture, out var result))
+			{
+				Logger?.Debug()
+					?.Log("Using provided transaction max spans `{ProvidedTransactionMaxSpans}' parsed as {ParsedTransactionMaxSpans}",
+						kv.Value, result);
+				return result;
```

Comment:
@SergeyKleyman, what do you think about validation for negative and zero value? I didn't find any limitations in other agents. And what about disable limitation? Does we need to support such case?

Reply:
I found that the only agent that specifies the expected behavior if negative value is provided is [Node.js](https://www.elastic.co/guide/en/apm/agent/nodejs/current/configuration.html#transaction-max-spans)
> Setting to -1 means that spans will never be dropped.

I can see why it can useful. I assume setting the option to zero causes agent not to send any spans at all. I would recommend going with the same approach (and documenting it)
- positive number N - agent drops spans above N
- `0` - agent drops all the spans
- `-1` - agent doesn't drops any spans
- any other negative value - invalid and agent falls back on the default value

Reply:
Implemented